morhendos/dontsign
# Add GA4 event tracking for file uploads and analysis

## Overview
This PR adds comprehensive GA4 event tracking to monitor user interactions with file uploads and document analysis.

## Changes
- Added event tracking for file selection
- Added event tracking for analysis start/completion
- Added error event tracking with specific error codes
- Added documentation for GA4 implementation

## Testing Instructions
1. Upload a valid file and check GA4 realtime for 'file_selected' event
2. Start analysis and verify 'analysis_started' event
3. Complete analysis and verify 'analysis_completed' event
4. Try invalid file to verify error tracking

## Tracked Events
### Engagement Events:
- file_selected
- analysis_started
- analysis_completed

### Error Events:
- file_error
- analysis_error

## Documentation
Added detailed documentation in `/docs/analytics.md` covering:
- Event categories and parameters
- Integration points
- Testing procedures
- Dashboard setup recommendations

## Notes
- All events include relevant metadata (file type, size)
- Error tracking is categorized by error type
- No breaking changes to existing functionality

## Checklist
- [x] Added event tracking code
- [x] Added error tracking
- [x] Verified in GA4 dashboard
